test/sparse/sparse.jl

```diff
@@ -1,469 +1,518 @@
+        @test all([se33 se33] == sparse([1, 2, 3, 1, 2, 3], [1, 2, 3, 4, 5, 6], ones(6)))
+    end
+
+    @testset "Vertical concatenation" begin
+        @test all([se33; se33] == sparse([1, 4, 2, 5, 3, 6], [1, 1, 2, 2, 3, 3], ones(6)))
+        se33_32bit = convert(SparseMatrixCSC{Float32,Int32}, se33)
+        @test all([se33; se33_32bit] == sparse([1, 4, 2, 5, 3, 6], [1, 1, 2, 2, 3, 3], ones(6)))
+    end
+
+    se44 = speye(4)
+    sz42 = spzeros(4, 2)
+    sz41 = spzeros(4, 1)
+    sz34 = spzeros(3, 4)
+    se77 = speye(7)
+    @testset "h+v concatenation" begin
+        @test all([se44 sz42 sz41; sz34 se33] == se77)
+    end
+
+    @testset "Blkdiag concatenation" begin
+        @test all(blkdiag(se33, se33) == sparse([1, 2, 3, 4, 5, 6], [1, 2, 3, 4, 5, 6], ones(6)))
+    end
+
+    @testset "Concatenation promotion" begin
+        sz41_f32 = spzeros(Float32, 4, 1)
+        se33_i32 = speye(Int32, 3, 3)
+        @test all([se44 sz42 sz41_f32; sz34 se33_i32] == se77)
+    end
+
+    @testset "Mixed sparse-dense concatenation" begin
+        sz33 = spzeros(3, 3)
+        de33 = eye(3)
+        @test  all([se33 de33; sz33 se33] == Array([se33 se33; sz33 se33 ]))
+    end
+
+        # check splicing + concatenation on
+        # random instances, with nested vcat
+        # also side-checks sparse ref
+    @testset "Splicing + concatenation on random instances" begin
+        for i = 1 : 10
+            a = sprand(5, 4, 0.5)
+            @test all([a[1:2,1:2] a[1:2,3:4]; a[3:5,1] [a[3:4,2:4]; a[5:5,2:4]]] == a)
+        end
+    end
+end
+
 a116 = copy(reshape(1:16, 4, 4))
 s116 = sparse(a116)
-p = [4, 1, 2, 3, 2]
-@test Array(s116[p,:]) == a116[p,:]
-@test Array(s116[:,p]) == a116[:,p]
-@test Array(s116[p,p]) == a116[p,p]
-
-# sparse assign
-p = [4, 1, 3]
-a116[p, p] = -1
-s116[p, p] = -1
-@test a116 == s116
-
-p = [2, 1, 4]
-a116[p, p] = reshape(1:9, 3, 3)
-s116[p, p] = reshape(1:9, 3, 3)
-@test a116 == s116
-
-# squeeze
-for i = 1:5
-    am = sprand(20, 1, 0.2)
-    av = squeeze(am, 2)
-    @test ndims(av) == 1
-    @test all(av.==am)
-    am = sprand(1, 20, 0.2)
-    av = squeeze(am, 1)
-    @test ndims(av) == 1
-    @test all(av.'.==am)
+
+@testset "Sparse ref" begin
+    p = [4, 1, 2, 3, 2]
+    @test Array(s116[p,:]) == a116[p,:]
+    @test Array(s116[:,p]) == a116[:,p]
+    @test Array(s116[p,p]) == a116[p,p]
+end
+
+@testset "Sparse assign" begin
+    p = [4, 1, 3]
+    a116[p, p] = -1
+    s116[p, p] = -1
+    @test a116 == s116
+
+    p = [2, 1, 4]
+    a116[p, p] = reshape(1:9, 3, 3)
+    s116[p, p] = reshape(1:9, 3, 3)
+    @test a116 == s116
+end
+
+@testset "Squeeze" begin
+    for i = 1:5
+        am = sprand(20, 1, 0.2)
+        av = squeeze(am, 2)
+        @test ndims(av) == 1
+        @test all(av.==am)
+        am = sprand(1, 20, 0.2)
+        av = squeeze(am, 1)
+        @test ndims(av) == 1
+        @test all(av.'.==am)
+    end
 end
 
 # matrix-vector multiplication (non-square)
```

Comment:
Perhaps preserve the "non-square" (or instead "rectangular") in the testset description and remove the then-redundant comment?